duckduckgo/iOS
# extract main view controller segues into extension

Task/Issue URL: https://app.asana.com/0/0/1205584556126566/f
Tech Design URL:
CC:

**Description**:

Extracts segues in the main view controller into an extension which is more programatic about opening what used to be storyboard references / segues.

Main.storyboard now only contains MainViewController

SuggestionTray was moved to a new storyboard

All segues were extracted to an extension 


**Steps to test this PR**:

* When testing, please compare with the production app.  There should be no visible difference. 
* Please test iPhone, iPad and all supported versions of iOS

Onboarding
1. Clean install
2. Launch app
3. Run through onboarding

Home Row prompt
1. Edit `showNow()` in `HomeRowReminder` to return true
2. Perform a search
3. Click on the prompt that appears  

Bookmarks
1. Launch bookmarks from the app's Home Screen
2. Browse to a page, launch bookmarks from the menu
3. Bookmark a page and then edit the bookmark from the menu
4. Favorite a page, return to the app's Home Screen and edit the favorite

Dax Action Sheets
1. Clean install of the app
2. Get through onboarding
3. Visit cnn.com -> Dax will tell you that trackers are being blocked, high five!
4. FireButton should be highlighted with pulsing circle
5. Tap fire button -> Dax will tell you about person data build up
6. Clear tabs and data
7. Perform a search -> Dax will tell you your searches are anonymous 
8. Visit Wikipedia -> No Dax dialog

Report Broken Site
1. Visit a website
2. Choose Report Broken Site

Downloads
1. Choose Downloads from the menu

Tab Switcher
1. Tap on the tabs button

Settings
1. Tap on settings from Home Screen 
2. Tap on settings from the menu
3. Turn on autofill in settings
4. Turn on cookie management 
5. Try and login in to Netflix.com and save the account
6. When the toast appears tap view 
7. Visit twitter.com
8. After cookie management kicks in open the privacy dashboard
9. Tap on Cookies Managed
10. Tap on disable in settings


**Orientation Testing**:


**Device Testing**:


**OS Testing**:


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
